PL2/PL2-shared/src/main/java/nl/tudelft/pl2016gr2/model/StringSequenceNode.java

```diff
@@ -0,0 +1,165 @@
+package nl.tudelft.pl2016gr2.model;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+/**
+ * A simple implementation of <code>Node</code> that represents the DNA sequence as String.
+ *
+ * @author Wouter Smit
+ */
+public class StringSequenceNode extends AbstractNode {
+
+  private String sequence;
+  private HashSet<String> genomes;
+  private HashSet<Integer> inEdges;
+  private HashSet<Integer> outEdges;
```

Comment:
Currently the in and out edges are only used in loops (contains isn't called on them, besides assert statements). I also don't really see any reason to call contains on a set of edges, so we could implement this as an `ArrayList` and use the `trimToSize()` method when all of the edges have been added. This will ensure that the `ArrayList` doesn't use more space than is required (it removes the array entries which aren't used).

The `HashSet` uses: `32 * SIZE + 4 * CAPACITY` bytes (where `SIZE` is the amount of objects stored in the `HashSet` and `CAPACITY` is the total capacity of the `HashSet`), see: http://java-performance.info/memory-consumption-of-java-data-types-2/.
If you use an `ArrayList` with a call to `trimToSize()` instead, the amount of memory needed will be: `SIZE` bytes.

There are 131 777 links and 94 258 nodes in the graph of 328 genomes. Assuming every node has fewer than 9 in links and fewer than 9 out links, this would mean that every hashset uses the default capacity of 16.
With these assumptions, the memory usage for the hashset implementation will be:
`(32 * 131 777 + 4 * 16 * 94 258) * 2 = 20MB`
With the arraylist implementation this would be only:
`(4 * 131 777) * 2 = 1MB`

There isn't really any downside to using an `ArrayList`, except that the `trimToSize()` method must be called after all of the in and out links have been added.

20MB might not sound like much, but this will probably be over 20% of the graph size after all of the other memory improvements have been applied and the memory reduction will apply to all of the subgraphs which are made. If we load the graph of 6000 genomes (or more) it will of course save a lot more than 20MB for each graph aswell.